pkg/cmd/util/clientcmd/factory.go

```diff
@@ -166,6 +169,31 @@ func NewFactory(clientConfig kclientcmd.ClientConfig) *Factory {
 	w.Printer = func(mapping *meta.RESTMapping, noHeaders, withNamespace, wide bool, showAll bool, columnLabels []string) (kubectl.ResourcePrinter, error) {
 		return describe.NewHumanReadablePrinter(noHeaders, withNamespace, wide, showAll, columnLabels), nil
 	}
+	kAttachablePodForObjectFunc := w.Factory.AttachablePodForObject
+	w.AttachablePodForObject = func(object runtime.Object) (*api.Pod, error) {
+		_, kc, err := w.Clients()
+		if err != nil {
+			return nil, err
+		}
+		switch t := object.(type) {
+		case *deployapi.DeploymentConfig:
+			var pods *api.PodList
```

Comment:
For this var, do you intend to retrieve the deployer pod or the first pod of the first deployment's RC?  As of now, the intent for `oc logs dc/foo` is to show the deployer pod's logs.

@kargakis if you have opinions.